src/animation.rs

```diff
@@ -0,0 +1,151 @@
+        self.time += delta_time;
+        for mut track in &mut self.clip.tracks {
+            match track.target {
+                keyframe::Target::Orientation(ref frames) => {
+                    let orientation = frames.interpolate(self.time);
+                    track.object.set_orientation(orientation);
+                },
+                keyframe::Target::Position(ref frames) => {
+                    let position = frames.interpolate(self.time);
+                    track.object.set_position(position);
+
+                },
+                keyframe::Target::Scale(ref frames) => {
+                    let scale = frames.interpolate(self.time);
+                    track.object.set_scale(scale);
+                },
+            }
+        }
+    }
+}
+
+/// A reusable set of keyframe tracks which represent an animation.
+#[derive(Clone, Debug)]
+pub struct Clip {
+    /// A name for this clip.
+    pub name: Option<String>,
+
+    /// The animation keyframe tracks.
+    pub tracks: Vec<keyframe::Track>,
+}
+
+pub mod keyframe {
+    use cgmath;
+    use mint;
+    use super::Interpolation;
+    use Object;
+
+    /// The frames of a target.
+    #[derive(Clone, Debug)]
+    pub struct Frames<T> {
+        pub times: Vec<f32>,
+        pub values: Vec<T>,
+    }
+
+    impl Frames<mint::Quaternion<f32>> {
+        pub fn interpolate(&self, time: f32) -> mint::Quaternion<f32> {
+            let t = time % self.times.last().unwrap();
+ 
+            let mut i = 0;
+            while t < self.times[i] {
+                i += 1;
+            }
+
+            let frame_start: cgmath::Quaternion<f32> = self.values[i].into();
+            let frame_end: cgmath::Quaternion<f32> = self.values[i + 1].into();
+            let s = t - self.times[i];
+            frame_start.slerp(frame_end, s).into()
+        }
+    }
+
+    impl Frames<mint::Point3<f32>> {
+        pub fn interpolate(&self, time: f32) -> mint::Point3<f32> {
+            let t = time % self.times.last().unwrap();
+
+            let mut i = 0;
+            while t < self.times[i] {
+                i += 1;
+            }
+
+            use cgmath::{EuclideanSpace, InnerSpace};
+            let frame_start = cgmath::Point3::<f32>::from(self.values[i]).to_vec();
+            let frame_end = cgmath::Point3::<f32>::from(self.values[i + 1]).to_vec();
+            let s = t - self.times[i];
+            cgmath::Point3::from_vec(frame_start.lerp(frame_end, s)).into()
+        }
+    }
+
+    impl Frames<f32> {
+        pub fn interpolate(&self, time: f32) -> f32 {
+            let t = time % self.times.last().unwrap();
+
+            let mut i = 0;
+            while t < self.times[i] {
+                i += 1;
+            }
+
+            let frame_start = self.values[i];
+            let frame_end = self.values[i + 1];
+            let s = t - self.times[i];
+            frame_start * (1.0 - s) + frame_end * s
+        }
+    }
+
+    #[derive(Clone, Debug)]
+    pub enum Target {
```

Comment:
three.js uses `PropertyBinding` and is a bit more flexible in terms of what the animation can target. I've made this enum for simplicity.